.k8s/search-deployment.yml

```diff
@@ -0,0 +1,143 @@
+# Search Service deployment that creates a container for the app.
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: search-deployment
+  labels:
+    app: search
+spec:
+  selector:
+    matchLabels:
+      app: search
+  template:
+    metadata:
+      labels:
+        app: search
+    spec:
+      containers:
+        - name: search
+          image: registry.default.gsp.test:80/cnxta/ion-search
+          imagePullPolicy: Always
+          volumeMounts:
+            - name: search-configs
+              mountPath: /etc/search-config
+              readOnly: true
+          ports:
+            - containerPort: 8080
+              protocol: TCP
+            - containerPort: 10052
+              protocol: TCP
+          args:
+            - "--spring.config.additional-location=file:/etc/search-config/k8s_search_config.yml"
+      volumes:
+        - name: search-configs
+          configMap:
+            name: search-config-map
+            defaultMode: 0755
+
+  # Solr deployment that creates a pod with one container and mounts the Persistent Volume Claim to it.
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: solr-deployment
+  labels:
+    app: solr
+spec:
+  selector:
+    matchLabels:
+      app: solr
+  template:
+    metadata:
+      labels:
+        app: solr
+    spec:
+      containers:
+        - name: solr
+          image: registry.default.gsp.test:80/cnxta/search-solr
+          imagePullPolicy: Always
+          resources:
+            limits:
+              cpu: "1"
+              memory: "4Gi"
+          ports:
+            - containerPort: 8983
+              protocol: TCP
+          volumeMounts:
+            - name: solr-pvc
+              mountPath: /var/solr/data
+      volumes:
+        - name: solr-pvc
```

Comment:
Nit-picking, but this is the persistent volume itself, not the pvc. This might be better as `solr-pv`.

Reply:
Good catch!